text/0020-contributing-guidelines.md

```diff
@@ -0,0 +1,46 @@
+# Contributing guidelines
+- Start Date: 2019/11/15
+- RFC PR: https://github.com/openmrs/openmrs-rfc-frontend/pull/20
+
+## Decision, including impact on distributions
+For frontend projects within OpenMRS, the following are used to instruct contributors on how to make a pull request.
+
+- [Github contributing.md files](https://help.github.com/en/github/building-a-strong-community/setting-guidelines-for-repository-contributors#adding-a-contributing-file)
+- [Github pull_request_template.md files](https://help.github.com/en/github/building-a-strong-community/creating-a-pull-request-template-for-your-repository)
+- [The OpenMRS code of conduct](https://wiki.openmrs.org/display/docs/Code+of+Conduct)
+- [The OpenMRS MPL-2.0 + Health care disclaimer software license](https://wiki.openmrs.org/display/RES/OpenMRS+License+FAQ)
+
+The [OpenMRS Pull Request Tips](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) are recommended.
+
+[Github squash and merge](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-request-merges#squash-and-merge-your-pull-request-commits) is enabled for all frontend projects.
+
+For frontend projects within OpenMRS, the following guidelines apply:
+- Not all PRs have to link to a JIRA ticket.
```

Comment:
“... though existing Jira tickets addressed by the PR must be linked.”?

Reply:
I think it's nice to link to a jira ticket when one exists. However, I think enforcing little things like the PR description discourage contribution and increase friction.

Reply:
If a project wants to encourage people to link to a JIRA ticket, that should go into the pull request template. Doesn't need to be in this RFC I think.

Reply:
Yeah, I don't think it should be solely on the submitter -- no one should "refuse reviewing." But we should collectively try and make sure Jira and Issues and PRs are well-linked, so we don't end up with a lot of detritus in the issue trackers. They're not useful if they're not tracking the issues.

Reply:
Just to make sure that i understand this correctly, for pull requests which have jira tickets, does including the link become optional?

Reply:
> for pull requests which have jira tickets, does including the link become optional?

This PR is scoped only to the microfrontends projects, but yes for those repos it is optional. A pull_request_template.md file can encourage the author to do so, and I think it's a good idea to link to a JIRA ticket when one exists. However, I don't think we should refuse to review a PR if it doesn't have a link to a JIRA ticket.

Reply:
So would you merge/close a pull request, which has a corresponding JIRA ticket, but where the pull request author has not linked the JIRA ticket?

Reply:
To me this is about encouraging progress by reducing the number of process-related hurdles in between the contributor and their code being merged. At other organizations I'm involved with, they usually link to a github issue or jira ticket when it exists. But they don't refuse to review the PR when that link is missing.

For me, this PR is not about identifying exactly which process has to be followed in every scenario -- it's about encouraging progress in the code. Right now the microfrontends code is under heavy feature development and the code is new, which is very different from many of the other code bases within OpenMRS. When something is new and under heavy development, it is common for a contributor just to see a thing that needs to happen and then write code to do it. As a project transitions from nascent to maintenance-only, then emphasizing process over progress makes more sense I think.